src/v/archival/ntp_archiver_service.cc

```diff
@@ -271,10 +271,30 @@ const cloud_storage::partition_manifest& ntp_archiver::manifest() const {
 
 ss::future<> ntp_archiver::start() {
     if (_parent.get_ntp_config().is_read_replica_mode_enabled()) {
-        ssx::spawn_with_gate(
-          _gate, [this] { return sync_manifest_until_abort(); });
+        ssx::spawn_with_gate(_gate, [this] {
+            return sync_manifest_until_abort().then([this] {
+                if (!_as.abort_requested()) {
+                    vlog(
+                      _rtclog.error,
+                      "Sync loop stopped without an abort being requested. "
```

Comment:
It seems odd we request the user to do something via manual config manipulation if it's universally the right choice and we could do it programmatically too?

Reply:
Right. In theory, it's not possible to end up there, but asserting out feels too aggressive. This log line is more for alerting than user instructions, but I tried to merge the two in one log line.